dynamic-tools/zap-extensions/zapwso2jiraplugin/ExtensionJiraIssueCreater.java

```diff
@@ -0,0 +1,199 @@
+/*
+ *
+ *  * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *  *
+ *  * WSO2 Inc. licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except
+ *  * in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing,
+ *  * software distributed under the License is distributed on an
+ *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  * KIND, either express or implied.  See the License for the
+ *  * specific language governing permissions and limitations
+ *  * under the License.
+ *
+ */
+
+package org.zaproxy.zap.extension.zapwso2jiraplugin;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+import org.json.JSONObject;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.zaproxy.zap.extension.api.API;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+import javax.naming.AuthenticationException;
+import javax.swing.ImageIcon;
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.jar.Manifest;
+
+/*
+ * An extension to create jira issues from alerts from the current session.
+ * This class is defines the extension.
+ */
+public class ExtensionJiraIssueCreater extends ExtensionAdaptor {
```

Comment:
We should be able to have the class name as "JiraIssueCreator" or "JiraIssueCreatorExtension"